src/init.cpp

```diff
@@ -1276,6 +1276,69 @@ bool AppInitMain(NodeContext& node, interfaces::BlockAndHeaderTipInfo* tip_info)
                                      chainman, *node.mempool, ignores_incoming_txs);
     RegisterValidationInterface(node.peerman.get());
 
+    // Check port numbers
+    const bilingual_str port_error = _("Invalid port specified in ");
+    if (args.IsArgSet("-port")) {
+        int64_t portIn = args.GetIntArg("-port", 0);
+        if (portIn <= 0 || portIn > std::numeric_limits<uint16_t>::max())
+            return InitError(strprintf(port_error + Untranslated("%s: %d"),"-port", portIn));
+    }
+    if (args.IsArgSet("-rpcport")) {
+        int64_t portIn = args.GetIntArg("-rpcport", 0);
+        if (portIn <= 0 || portIn > std::numeric_limits<uint16_t>::max())
+            return InitError(strprintf(port_error + Untranslated("%s: %d"),"-rpcport", portIn));
+    }
+
+    std::string hostOut;
+    uint16_t portOut{0};
+    for (const std::string& socketAddr : args.GetArgs("-bind")) {
```

Comment:
A lot of repetition. Maybe an outer for loop over each option name?

Reply:
Done, good point